File: C/exo3_3.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main()
{
    char fileName[200];
    printf("Nom du fichier texte: ");
    scanf("%s", fileName);
    
    if(access(fileName, F_OK)!=0)  //si le fichier n'exsite pas
    {
        printf("Le fichier n'existe pas.\n");
        exit(1);
    }
    char *argv[]={"emacs", fileName, NULL};
    execv("/usr/bin/emacs", argv);
    return 0;
}

File: python/exo3_3.py
import os
import sys

fileName = input("Nom du fichier texte: ")

if not os.access(fileName, os.F_OK) :
        print("Le fichier n'existe pas.")
        sys.exit(1)

argv = ["emacs", fileName]
os.execv("/usr/bin/emacs", argv)